odoo/addons/base/models/avatar_mixin.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from hashlib import sha512
+from base64 import b64encode
+
+from odoo import models, fields, api
+
+from odoo.tools import html_escape, file_open
+from odoo.modules.module import get_resource_path
+
+
+class AvatarMixin(models.AbstractModel):
+    _name = 'avatar.mixin'
+    _inherit = ['image.mixin']
+    _description = "Avatar Mixin"
+
+    # all image fields are base64 encoded and PIL-supported
+
+    avatar_1920 = fields.Image("Avatar", max_width=1920, max_height=1920, compute="_compute_avatar_1920", readonly=False)
+
+    avatar_1024 = fields.Image("Avatar 1024", max_width=1024, max_height=1024, compute="_compute_avatar_1024", readonly=False)
+    avatar_512 = fields.Image("Avatar 512", max_width=512, max_height=512, compute="_compute_avatar_512", readonly=False)
+    avatar_256 = fields.Image("Avatar 256", max_width=256, max_height=256, compute="_compute_avatar_256", readonly=False)
+    avatar_128 = fields.Image("Avatar 128", max_width=128, max_height=128, compute="_compute_avatar_128", readonly=False)
+
+    def _compute_avatar(self, avatar_field, image_field):
+        for record in self:
+            record[avatar_field] = record[image_field] or record._get_one_avatar(avatar_field, image_field)
+
+    def _get_one_avatar(self, avatar_field, image_field):
+        self.ensure_one()
+        if self[self._rec_name]:
+            return self._generate_avatar_svg(self[self._rec_name])
+        return self._get_avatar_placeholder()
+
+    @api.depends(lambda self: [self._rec_name, 'image_1920'])
+    def _compute_avatar_1920(self):
+        self._compute_avatar('avatar_1920', 'image_1920')
+
+    @api.depends(lambda self: [self._rec_name, 'image_1024'])
+    def _compute_avatar_1024(self):
+        self._compute_avatar('avatar_1024', 'image_1024')
+
+    @api.depends(lambda self: [self._rec_name, 'image_512'])
+    def _compute_avatar_512(self):
+        self._compute_avatar('avatar_512', 'image_512')
+
+    @api.depends(lambda self: [self._rec_name, 'image_256'])
+    def _compute_avatar_256(self):
+        self._compute_avatar('avatar_256', 'image_256')
+
+    @api.depends(lambda self: [self._rec_name, 'image_128'])
+    def _compute_avatar_128(self):
+        self._compute_avatar('avatar_128', 'image_128')
+
+    def _generate_avatar_svg(self, name):
+        if not name:
+            return self._get_avatar_placeholder()
+        base = sha512(name.encode('utf-8'))
```

Comment:
Also can't we use `create_date` now?
In partner and employee this can be override to use matching internal user create_date to keep the same color.
(Then you need to move the getting of the value it in a method to be able to override it)